tests/Oryx.Integration.Tests/Php/PhpFpmConfigTest.cs

```diff
@@ -0,0 +1,226 @@
+        [InlineData("false", "5", "5", "5", "pm.max_children must be a positive value")]
+        [InlineData("-1", "5", "5", "5", "pm.max_children must be a positive value")]
+        [InlineData("10", "5", "13", "12", "pm.min_spare_servers(12) and pm.max_spare_servers(13) cannot be greater than pm.max_children(10)")]
+        [InlineData("10", "9", "8", "7", "pm.start_servers(9) must not be less than pm.min_spare_servers(7) and not greater than pm.max_spare_servers(8)")]
+        public async Task PipelineTestFailInvocationsPhp80(
+            string fpmMaxChildren,
+            string fpmStartServers,
+            string fpmMaxSpareServers,
+            string fpmMinSpareServers,
+            string failureOutputText)
+        {
+            await PhpFpmConfigTestFailuresAsync(
+                "8.0",
+                fpmMaxChildren,
+                fpmStartServers,
+                fpmMaxSpareServers,
+                fpmMinSpareServers,
+                failureOutputText);
+        }
+
+        [Theory, Trait("category", "php-7.4")]
+        [InlineData("10", "10", "7", "7", "8", "8", "6", "6")]
+        [InlineData("", "5", "", "2", "", "3", "", "1")] // defaults
+        public async Task PipelineTestInvocationsPhp74(
+            string fpmMaxChildren, string expectedFpmMaxChildren,
+            string fpmStartServers, string expectedFpmStartServers,
+            string fpmMaxSpareServers, string expectedFpmMaxSpareServers,
+            string fpmMinSpareServers, string expectedFpmMinSpareServers)
+        {
+            await PhpFpmConfigTestAsync(
+                "7.4",
+                fpmMaxChildren, expectedFpmMaxChildren,
+                fpmStartServers, expectedFpmStartServers,
+                fpmMaxSpareServers, expectedFpmMaxSpareServers,
+                fpmMinSpareServers, expectedFpmMinSpareServers);
+        }
+
+        [Theory, Trait("category", "php-7.4")]
+        [InlineData("false", "5", "5", "5", "pm.max_children must be a positive value")]
+        [InlineData("-1", "5", "5", "5", "pm.max_children must be a positive value")]
+        [InlineData("10", "5", "13", "12", "pm.min_spare_servers(12) and pm.max_spare_servers(13) cannot be greater than pm.max_children(10)")]
+        [InlineData("10", "9", "8", "7", "pm.start_servers(9) must not be less than pm.min_spare_servers(7) and not greater than pm.max_spare_servers(8)")]
+        public async Task PipelineTestFailInvocationsPhp74(
+            string fpmMaxChildren, 
+            string fpmStartServers,
+            string fpmMaxSpareServers,
+            string fpmMinSpareServers,
+            string failureOutputText)
+        {
+            await PhpFpmConfigTestFailuresAsync(
+                "7.4",
+                fpmMaxChildren,
+                fpmStartServers,
+                fpmMaxSpareServers,
+                fpmMinSpareServers,
+                failureOutputText);
+        }
+
+        private async Task PhpFpmConfigTestAsync(
```

Comment:
nit: it looks like this method and `PhpFpmConfigTestFailuresAsync` share a lot of the same code -- I think it could be worth looking into consolidating them into a single method that have the same arguments with an additional optional argument for the `failureOutputText` whose existence changes the assertion portion of the code.